datafusion/core/src/datasource/file_format/arrow.rs

```diff
@@ -298,7 +300,7 @@ impl FileSink for ArrowFileSink {
         demux_task
             .join_unwind()
             .await
-            .map_err(|e| DataFusionError::ExecutionJoin(Box::new(e)))??;
+            .map_err(|e| exec_datafusion_err!("ExecutionJoin error: {}", e))??;
```

Comment:
I'm still quite new to the codebase, but I think using `DataFusionError::Execution` here is not equivalent to `DataFusionError::ExecutionJoin`. 

There are other places we used `exec_datafusion_err!` for `ExecutionJoin` as well.

Reply:
correct